### PR TITLE
Update ref_name-description.md

### LINKED ISSUE #15319: GITHUB_REF and GITHUB_REF_NAME examples in Actions docs unclear
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables

### What part(s) of the article would you like to see updated?

`GITHUB_REF` and `GITHUB_REF_NAME` show example values of `refs/heads/feature-branch-1` and `feature-branch-1` respectively. However this is inconsistent with the values populated in the case of a PR-triggered workflow. 

Given the description and example, one would expect to see the branch that triggered the workflow when running a PR-triggered workflow. Instead these variables are populated by values like `refs/pull/26/merge` and `26/merge` respectively. The variable that contains the desired data in this case is actually `GITHUB_HEAD_REF`.

The documentation should be updated to more accurately describe the content of GitHub Actions predefined variables.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by antoinegomez**
### Why:

The actual documentation is not complete. The value is not the same when the action is triggered by a pull_request. This create confusion and makes people loose time.

Closes: https://github.com/github/docs/issues/15319

### Check off the following:

- [ ] I have reviewed my changes in staging, available via the **View deployment** link in this PR's timeline.

  - For content changes, you will also see an automatically generated comment with links directly to pages you've modified. The comment won't appear if your PR only edits files in the `data` directory.
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Review by jc-clark — COMMENTED** [review]
Thank you for helping with this clarification, @antoinegomez! 

The line breaks in this reusable file break the table in the "Contexts" article. You can see this in the rendered preview environment here: 

https://docs-29712-d2378a.preview.ghdocs.com/en/actions/learn-github-actions/contexts

I added a couple `<br>` breaks to fix this and suggested a slight change in wording for clarity. Let me know what you think of this!

File: data/reusables/actions/ref_name-description.md
Comment (antoinegomez) [reply data/reusables/actions/ref_name-description.md:3]: Thanks for your reply, yes this looks good for me.

**Review by jc-clark — APPROVED** [review]
All looks good to me @antoinegomez. Our documentation repositories are currently frozen, so we can't merge this at the moment. But I added the `merge-after-freeze` label here so we can merge this once we thaw the repositories (which will be at the end of this week). Thanks! :shipit:

**Comment by aspiers** [comment]
Either this PR was erroneous, or the behaviour has changed since it was documented, because now `github.ref_name` returns values like `5/merge`.  I suspect the former, because I see that this refers to #15319, which was originally reported way back in Feb 2022 and when referring to `github.ref` and `github.ref_name` it states: 

> Instead these variables  are populated by values like `refs/pull/26/merge` and `26/merge` respectively.

I have submitted #30907 to fix this.

**Pull request merged**
